client/signup/steps/domains/index.jsx

```diff
@@ -486,6 +487,12 @@ class DomainsStep extends React.Component {
 				shouldHideFreeDomainExplainer={ shouldHideFreeDomainExplainer }
 			/>
 		);
+
+		if ( 'launch-site' === this.props.flowName ) {
```

Comment:
I wonder if we can just always add the `CartData` since we're clearing out the selectedSite in the controler?